### PR TITLE
refactor(cardinal): replace eventhub with socketio middleware

### PR DESCRIPTION
**Pull request by smsunarto**
Closes: WORLD-XXX

## Overview

> Description of the overall background and high-level changes that this PR introduces

This PR replaces the EventHub with Fiber's socketio middleware that significantly simplifies managing websocket connections (broadcasting, sending message to a particular client, etc) while being more robust.

## Brief Changelog

- Remove everything in `/cardinal/events`
- Simplify broadcasting events to WS connections is now only a single step of `server.BroadcastEvent` instead of having to `EmitEvent` then `FlushEvents`
- Fixed flaky test in `cardinal.TestShutdownViaSignal` arising from test terminating before goroutine finished executing.
- Remove `TestEvents` as it is redundant against `TestEventsThroughSystems`, which allows us to eliminate `world.GetEventHub`

## Testing and Verifying

- All existing passes

### CODE REVIEW

**Comment by smsunarto** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-engine/690?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#690** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-engine/690?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#689** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-engine/689?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @smsunarto and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cardinal/server/server.go [hunk 1]
```diff
@@ -76,19 +71,41 @@ func (s *Server) Serve() error {
...
 	err = s.app.Listen(":" + s.config.port)
 	if err != nil {
 		return eris.Wrap(err, "error starting Fiber app")
 	}
-	s.isRunning.Store(false)
+
+	return nil
+}
+
+func (s *Server) BroadcastEvent(event any) error {
```
Comment (technicallyty) [inline cardinal/server/server.go:85]: this name feels a bit misleading; reading this code id assume we are broadcasting a singular event, but its only usage is broadcasting tick results which contains multiple events + some other data
```
type TickResults struct {
    Tick     uint64
    Receipts []receipt.Receipt
    Events   [][]byte
}
```

unless the intent is that `TickResults` is itself to be considered an `event`. unsure! but seems a bit quirky

Comment (smsunarto) [reply cardinal/server/server.go:85]: im equally unsure, im just following what was there. i say we just keep it this way then revisit

**Review by technicallyty — APPROVED** [review]

File: cardinal/server/server.go (hunk 1 above)
Comment (pyrofolium) [reply cardinal/server/server.go:85]: Anything json serializeable is an event. The eventHub was originally agnostic to what a tick is or what the game engine is, it's just something that broadcasts events. 

I agree with @smsunarto keeping the event hub as it is allows for future extension. EventHub right now is only used to send TickResults over one websocket connection, but it supports much more. Well it's not an eventHub anymore but the intent is to allow the broadcasting of anything.

**Pull request merged**